src/lib/containers/download_list/DownloadConfirmation.tsx

```diff
@@ -120,64 +139,86 @@ const StatusConstruct: UiStateDictionary = {
+     // is not defined (configured for a container)
   const [showDownloadList, setShowDownloadList] = useState(false)
-  const queryBundleRequest = getLastQueryRequest!()
+  
+  const updateStats = useCallback(async (count?: number, bytes?: number) => {
+    if (accessToken) {
+      const estimatedDownloadBytesPerSecond = await testDownloadSpeed(accessToken)
+      const size = bytes ?? 0
+      const fileCount = count ?? 0
+      const durationSeconds = size / estimatedDownloadBytesPerSecond
+      const duration = moment.duration(durationSeconds, 'seconds').humanize()
+      setStatus(StatusEnum.INFO)
+      setState({
+        fileCount: fileCount,
+        fileSize: size,
+        downloadEstimate: duration,
+      })
+    }
+  }, [accessToken])
 
-  const getFilesInformation = async (
-    queryBundleRequest: QueryBundleRequest,
+  const {
+    data,
+    isSuccess
+  } = useGetEntityChildren (
+    {
+      parentId: queryOrFolder.folderId,
+      includeSumFileSizes: true,
+      includeTotalChildCount: true,
+      includeTypes: [EntityType.FILE]
+    }
+  )
+  useEffect(() => {
+    if (isSuccess && data) {
+      updateStats(data.totalChildCount, data.sumFileSizesBytes)
+    }
+  }, [updateStats, queryOrFolder, isSuccess, data])
+
+  const getFilesInformation = useCallback(async (
     token: string,
   ) => {
-    setState({
-      ...state,
-      status: StatusEnum.LOADING_INFO,
-    })
+    setStatus(StatusEnum.LOADING_INFO)
 
     const partMask =
-      SynapseConstants.BUNDLE_MASK_QUERY_COUNT |
-      SynapseConstants.BUNDLE_MASK_SUM_FILES_SIZE_BYTES
+    SynapseConstants.BUNDLE_MASK_QUERY_COUNT |
+    SynapseConstants.BUNDLE_MASK_SUM_FILES_SIZE_BYTES
 
     const queryBundleRequestSizeInformation: QueryBundleRequest = {
-      ...queryBundleRequest,
+      ...queryOrFolder.queryBundleRequest!,
       partMask,
     }
-
-    const {
-      queryCount,
-      sumFileSizes,
-    } = await SynapseClient.getQueryTableResults(
+    const {queryCount, sumFileSizes} = await SynapseClient.getQueryTableResults(
       queryBundleRequestSizeInformation,
       token,
     )
-    const estimatedDownloadBytesPerSecond = await testDownloadSpeed(token)
-    const size = sumFileSizes ? sumFileSizes['sumFileSizesBytes'] : 0
-    const durationSeconds = size / estimatedDownloadBytesPerSecond
-    const duration = moment.duration(durationSeconds, 'seconds').humanize()
-    setState({
-      fileCount: queryCount || 0,
-      fileSize: size,
-      downloadEstimate: duration,
-      status: StatusEnum.INFO,
-    })
-  }
+    updateStats(queryCount, sumFileSizes?.sumFileSizesBytes)
+  }, [queryOrFolder, setStatus, updateStats])
 
   // UseEffect memoization only works for arguments where a direct === comparison can be made
   // This fails drastically with the queryBundleRequest object which is a complex object of many fields that
   // change, we could use a custom comparitor but this also introduces risk
   useDeepCompareEffect(() => {
```

Comment:
This is why this is a bit more complicated than I'd like.  The useDeepCompareEffect() requires a js Object (it will throw an error that tells you to instead use useEffect() if only primitives are provided).  In the case when the queryBundleRequest is undefined (if configured to add Folder/Project children), then this error is thrown at runtime :(
To avoid this, I created a new object that contains the folderId and queryBundleRequest, and define that in the dependencies for this hook.  Then everyone is happy :)